OpenCensusAgent/opencensus-service/receiver/opencensusreceiver/octrace/opencensus.go

```diff
@@ -90,6 +97,55 @@ var errTraceExportProtocolViolation = errors.New("protocol violation: Export's f
 
 const receiverTagValue = "oc_trace"
 
+type mdsdJSON struct {
+	Tag    string   `json:"TAG"` // Tag is used to see if acknowledged by 1Agent
+	Source string   `json:"SOURCE"`
+	Data   []string `json:"DATA"` // Data must be an array
+}
+
+// SendTraceTo1Agent makes a connection to 1Agent and passes in the OpenCensus proto trace in JSON format
+func SendTraceTo1Agent(td *data.TraceData) {
+	log.Println("Starting funnel of data.")
+
+	conn, err := net.Dial("unix", "/var/run/mdsd/default_json.socket")
+	if err != nil {
+		log.Printf("Error connecting: %v", err)
+		return
+	}
+	traceData, err := json.Marshal(td)
+	if err != nil {
+		log.Printf("Error marshaling trace data: %v", err)
+		return
+	}
+	dataList := []string{
+		string(traceData), //WholeTrace, this part is defined in schema found in mdsd.xml
+	}
+	id := time.Now().UTC()
+
+	trace := new(mdsdJSON)
+	trace.Tag = strconv.FormatInt(id.Unix(), 10) // Use time to create a unique Tag
```

Comment:
what is the time resolution? we want subsecond; otherwise, won't be unique.

Reply:
For my prototype, a trace will be sent every 3 seconds so this should not be a problem. I will address this in another PR if I have time this summer.